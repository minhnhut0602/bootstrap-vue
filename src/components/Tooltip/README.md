# Tooltip

```vue live
<Tooltip>Default Example Usage</Tooltip>
```

## Props

| Prop name    | Description | Type                         | Values | Default           |
| ------------ | ----------- | ---------------------------- | ------ | ----------------- |
| boundary     |             | string\|HTMLElement          | -      | 'clippingParents' |
| container    |             | string\|HTMLElement\|boolean | -      | false             |
| customClass  |             | string\|array\|object        | -      | ''                |
| delay        |             | number\|object               | -      | 0                 |
| html         |             | boolean                      | -      | false             |
| offset       |             | string \| number[]           | -      | () => [0, 0]      |
| popperConfig |             | object\|func                 | -      | () => ({})        |
| tag          |             | string                       | -      | 'div'             |
| title        |             | string                       | -      | ''                |

## Slots

| Name    | Description | Bindings |
| ------- | ----------- | -------- |
| default |             |          |
