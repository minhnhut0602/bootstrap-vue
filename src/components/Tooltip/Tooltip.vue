<template>
    <div
        ref="tooltip"
        class="d-inline-block"
    >
        <slot v-if="!!$slots.default" />
    </div>
</template>

<script lang="ts">
import {PropType, computed, ref} from 'vue';
import useBootstrapEmits from '@/composables/useBootstrapEmits';
import useBootstrapInstance from '@/composables/useBootstrapInstance';

const tooltipEvents = ['show', 'shown', 'hide', 'hidden', 'inserted'] as const;

type TooltipEvent = typeof tooltipEvents[number];
</script>

<script lang="ts" setup>
const props = defineProps({
    boundary: {
        type: [String, HTMLElement],
        default: 'clippingParents',
    },
    container: {
        type: [String, HTMLElement, Boolean],
        default: false,
    },
    customClass: {
        type: [String, Array, Object],
        default: '',
    },
    delay: {
        type: [Number, Object],
        default: 0,
    },
    fallbackPlacements: {
        type: Array as PropType<string[]>,
        default: () => ['top', 'right', 'bottom', 'left'],
    },
    html: {
        type: Boolean,
        default: false,
    },
    offset: {
        type: [String, Array],
        default: () => [0, 0],
    },
    placement: {
        type: String,
        default: 'top',
    },
    popperConfig: {
        type: [Object, Function],
        default: () => ({}),
    },
    title: {
        type: String,
        default: '',
    },
    trigger: {
        type: String,
        default: 'hover focus',
    },
});

const tooltip = ref<HTMLElement>();

const emit = defineEmits<{(event: TooltipEvent): void}>();

const tooltipClass = computed(() => {
    if (typeof props.customClass === 'string') {
        return props.customClass;
    }

    if (props.customClass.constructor === Array) {
        return props.customClass.join(' ');
    }

    return Object.keys(props.customClass)
        .filter((key: string) => !!props.customClass[key])
        .join(' ');
});

useBootstrapEmits(
    tooltip,
    tooltipEvents,
    emit,
    'tooltip',
);

const {bsInstance: bsTooltip} = useBootstrapInstance(
    'Tooltip',
    tooltip,
    {
        ...props,
        customClass: tooltipClass.value,
    },
);

defineExpose({bsTooltip});
</script>
