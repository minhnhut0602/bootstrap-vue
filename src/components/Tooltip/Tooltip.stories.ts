import Tooltip from '.';

export default {
    title: 'Components/Tooltip',
    component: Tooltip,
    argTypes: {},
};

const defaultArgs = {
};

const template = (args) => ({
    components: {Tooltip},
    setup: () => ({args}),
    template: '<Tooltip v-bind="args" />',
});

export const Default = template.bind({});

Default.args = defaultArgs;
