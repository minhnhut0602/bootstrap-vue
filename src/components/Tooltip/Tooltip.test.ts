import Tooltip from '@/components/Tooltip';

describe('template', () => {
    componentRenderTest(Tooltip);

    componentWrapperTagTest(Tooltip, {props: {tag: 'button'}});
});
