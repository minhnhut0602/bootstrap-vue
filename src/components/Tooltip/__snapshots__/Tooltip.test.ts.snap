// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template renders default 1`] = `"<div></div>"`;
